lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -14,9 +14,26 @@ namespace lldb_private {
 bool LLDBMemoryReader::queryDataLayout(DataLayoutQueryType type, void *inBuffer,
                                        void *outBuffer) {
   switch (type) {
-  // FIXME: add support for case DLQ_GetPtrAuthMask rdar://70729149
-  case DLQ_GetPtrAuthMask:
-    return false;
+  case DLQ_GetPtrAuthMask: {
+    ArchSpec arch = m_process.GetTarget().GetArchitecture();
+    if (!arch.IsValid())
+      return false;
+
+    if (arch.GetTriple().getVendor() != llvm::Triple::Apple ||
+        arch.GetTriple().getArch() != llvm::Triple::aarch64)
+      return false;
+
+    int number_of_addressable_bits =
+        m_process.GetAddressingBits().getValueOr(37);
+
+    int addressable_bits_setting = m_process.GetVirtualAddressableBits();
+    if (addressable_bits_setting != 0 && addressable_bits_setting < 63)
+      number_of_addressable_bits = addressable_bits_setting;
+
+    uint64_t mask_pattern = (1ULL << number_of_addressable_bits) - 1;
```

Comment:
I'm wondering (too late) if we can't use `Process::GetCodeAddressMask` or `Process::GetDataAddressMask` for this

Reply:
TIL, and emphatically yes!